kinesis/src/main/scala/com.snowplowanalytics.snowplow.collectors.scalastream/sinks/KinesisSink.scala

```diff
@@ -185,10 +185,10 @@ class KinesisSink private (
               s"Successfully wrote ${batch.size - failurePairs.size} out of ${batch.size} records to Kinesis stream $streamName."
             )
             if (failurePairs.nonEmpty) {
-              failurePairs.foreach(f =>
-                log.error(
-                  s"Failed writing record to Kinesis stream $streamName, with error code [${f._2.getErrorCode}] and message [${f._2.getErrorMessage}]."
-                )
+              val errorCodes = failurePairs.map(_._2.getErrorCode).toSet.mkString(",")
+              val exampleMsg = failurePairs.map(_._2.getErrorMessage).find(_.nonEmpty).getOrElse("")
+              log.error(
+                s"Failed writing ${failurePairs.size} records to Kinesis stream $streamName, with error codes [$errorCodes] and example message [$exampleMsg]."
```

Comment:
It might be more helpful if we have an example message for each error code, but not a big deal if you'd rather go ahead as is.

Reply:
You're completely right.  I just pushed an update.